src/osd/ECBackend.cc

```diff
@@ -1375,9 +1404,15 @@ int ECBackend::get_min_avail_to_read_shards(
   }
 
   set<int> need;
-  int r = ec_impl->minimum_to_decode(want, have, &need);
-  if (r < 0)
-    return r;
+  if (do_redundant_reads) {
+    if (have.size() < ec_impl->get_data_chunk_count())
```

Comment:
in the case of LRC, 
```
plugin=lrc \
             mapping=__DD__DD \
             layers='[
                       [ "_cDD_cDD", "plugin=isa technique=cauchy" ],
                       [ "cDDD____", "plugin=isa" ],
                       [ "____cDDD", "plugin=jerasure" ],
                     ]'
```
if we don't have chunk 2, which happens to be a data chunk. so in addition to 3, 6, 7, we need to read 0,1,3, to decode the chunks. where 

- `get_data_chunk_count()` returns 4
- need.size() should be len([3,6,7,0,1,3]), i.e. **6**
- `have.size()` could be [3,6,7,0,1], i.e. **5**. if we also miss chunk 1. please note that **1** is needed to reconstruct 2.

but your patch will not return `-EIO` in this scenario. as `have.size() >= get_data_chunk_count()`.